### ISSUE TITLE
jsf sheet does not display under TabView On Google Chrome and Mozilla Firefox

### ISSUE THREAD

**Issue by amolksagar** [marklcg/jsf-sheet#12]
Hi All,

    
[testfile.txt](https://github.com/marklcg/jsf-sheet/files/1010377/testfile.txt)

     I am displaying jsf sheet (latest jar created from 1.5 code) under p:tabView component of Primefaces. The issue comes with Google Chrome and Firefox where the lcg:sheet is not visible. If I try this on Internet Explorer (IE11) then then there is no issue in displaying the sheet. Also if I keep the sheet outside of the TabView Chrome/Firefox does not complain.

Attached is the xhtml. I am using Google Chrome Version 58.0.3029.81 and FireFox 53.0 (32-bit).Is there any extension which I need to install on Google Chrome/FireFox to support an embedded jsf sheet?

I dont see any error on the Chrome/FireFox Console in the Debugger tools.

Please help as I have been debugging this issue since a long time and came to the point where I can pin point on the combination (Chrome/FireFox+TabView+jsf Sheet) being a problem.

I can however see the sheet loaded in the DOM but not displayed. Attaching screenshot

![capture](https://cloud.githubusercontent.com/assets/19744503/26196985/08a6d96c-3c04-11e7-9efb-7aa8325fd1f6.png)

**Comment by amolksagar** [comment]
Hi All,

  Apparently the issue only comes when you dont mention the height attribute. Kept all the other code same as before and copy pasted a plain lcg:sheet from one of Mark's examples. It started working. On further analysis I could see that if you dont mention the height attribute in the lcg:sheet then the sheet wont appear in Google Chrome

**Comment by melloware** [comment]
That is interesting.  I always assign a height to my sheets so I have not noticed this behavior before.   You mentioned Firefox and Chrome but I assume this behavior of not assigning a height does not matter what browser you are using?

**Comment by amolksagar** [comment]
The issue comes only when you use a sheet inside a tab view. Sheets outside the tabview are working fine with or without the height attribute in any browser.

**Comment by marklcg** [comment]
Ok, spent time looking at this and the issue appears to be two-fold.  If Handsontable is not given a height, it will find the nearest parent with the overflow set to hidden and will use the height of that element.  The problem is that our inner div has no height specified so its 0.  If you don't have a header or footer on your table, it will appear it is not rendering at all, but is simply a 0 height.

So adding this style to your CSS will at least get the inner div to match the width and height of your containing element:

```css
.ui-handsontable, .handsontable-inner {
    height: 100%;
    width: 100%;
}
```
That said, handontable still requires some sort of container that will define the boundaries for the table.  See [this link](https://handsontable.com/blog/articles/a-complete-guide-to-changing-size-of-handsontable) for more on how it works.

The tab component doesn't have a styled height and in fact will ignore your styleClass and style attributes (perhaps a PF issue). So I recommend putting a div inside your tab and styling it with a height. 

```xml
<p:tab title="Assets">
    <div class="tabcontent">
        <lcg:sheet id="sheet" value="#{assets}" var="row" label="Sheet" rowKey="#{row.assetId}"
                               fixedCols="2" showRowHeaders="false" sortBy="#{row.assetId}" sortOrder="ascending">
```
```css
.tabcontent {
    height: 400px;
}
```
In short, at some level something has to set the height so that Handsontable can determine its size.

I will commit a fix for the CSS mentioned above, but for now you can add it to your CSS.

**Comment by amolksagar** [comment]
Thanks Melloware. Are you planning to release version 1.5 to Maven central with all the bugs resolved?

Regards.

**Comment by marklcg** [comment]
I do plan to release a version soon.

**Comment by marklcg** [comment]
Version 1.5 deployed which should contain the CSS fix.
